pkg/cluster/cluster.go

```diff
@@ -505,13 +505,17 @@ func (c *Cluster) reportFailedStatus() {
 	f := func() (bool, error) {
 		c.status.SetPhase(spec.ClusterPhaseFailed)
 		err := c.updateStatus()
-		if err == nil || k8sutil.IsKubernetesResourceNotFoundError(err) {
+		if err == nil {
 			return true, nil
 		}
-
+		// Update (PUT) will return conflict even if object is deleted.
+		// Because it will check UID first and return something like: "Precondition failed: UID in precondition: 0xc42712c0f0, UID in object meta: ".
 		if apierrors.IsConflict(err) {
 			cl, err := k8sutil.GetClusterTPRObject(c.config.KubeCli.Core().GetRESTClient(), c.cluster.Namespace, c.cluster.Name)
 			if err != nil {
+				if k8sutil.IsKubernetesResourceNotFoundError(err) {
```

Comment:
move the comment to here? talk about why get after conflict can return not found?